Vous êtes Data Scientist dans une start-up de la EdTech, nommée academy, qui propose des contenus de formation en ligne pour un public de niveau lycée et université.

Mark, votre manager, vous a convié à une réunion pour vous présenter le projet d’expansion à l’international de l’entreprise. Il vous confie une première mission d’analyse exploratoire, pour déterminer si les données sur l’éducation de la banque mondiale permettent d’informer le projet d’expansion.

Voici les différentes questions que Mark aimerait explorer, que vous avez notées durant la réunion :

Quels sont les pays avec un fort potentiel de clients pour nos services ?
Pour chacun de ces pays, quelle sera l’évolution de ce potentiel de clients ?
Dans quels pays l'entreprise doit-elle opérer en priorité ?
Votre mission
Mark vous a donc demandé de réaliser une analyse pré-exploratoire de ce jeu de données. Il vous a transmis cet email à la suite de la réunion :

Hello,

Les données de la Banque mondiale sont disponibles à l’adresse suivante :

https://datacatalog.worldbank.org/dataset/education-statistics

Ou en téléchargement direct à ce lien.

Je te laisse regarder la page d'accueil qui décrit le jeu de données. En résumé, l’organisme “EdStats All Indicator Query” de la Banque mondiale répertorie 4000 indicateurs internationaux décrivant l’accès à l’éducation, l’obtention de diplômes et des informations relatives aux professeurs, aux dépenses liées à l’éducation... Tu trouveras plus d'info sur ce site :

http://datatopics.worldbank.org/education/

Pour la pré-analyse, pourrais-tu :

Valider la qualité de ce jeu de données (comporte-t-il beaucoup de données manquantes, dupliquées ?)

Décrire les informations contenues dans le jeu de données (nombre de colonnes ? nombre de lignes ?)

Sélectionner les informations qui semblent pertinentes pour répondre à la problématique (quelles sont les colonnes contenant des informations qui peuvent être utiles pour répondre à la problématique de l’entreprise ?)

Déterminer des ordres de grandeurs des indicateurs statistiques classiques pour les différentes zones géographiques et pays du monde (moyenne/médiane/écart-type par pays et par continent ou bloc géographique)

Ton travail va nous permettre de déterminer si ce jeu de données peut informer les décisions d'ouverture vers de nouveaux pays. On va partager ton analyse avec le board, alors merci de soigner la présentation et de l'illustrer avec des graphiques pertinents et lisibles !

# CSV used : EdStatsData

## Libraries imports

In [82]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

## File imports

In [83]:
filename = 'EdStatsData.csv'
directory = '/CSV'
dfEdStatsData = pd.read_csv(os.getcwd() + directory + '/' + filename)

## Basic informations about the dataset

> Data sample :

In [84]:
dfEdStatsData.head()

Country Name Country Code  \
0   Arab World          ARB   
1   Arab World          ARB   
2   Arab World          ARB   
3   Arab World          ARB   
4   Arab World          ARB   

                                      Indicator Name  Indicator Code  \
0  Adjusted net enrolment rate, lower secondary, ...      UIS.NERA.2   
1  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.F   
2  Adjusted net enrolment rate, lower secondary, ...  UIS.NERA.2.GPI   
3  Adjusted net enrolment rate, lower secondary, ...    UIS.NERA.2.M   
4  Adjusted net enrolment rate, primary, both sex...     SE.PRM.TENR   

        1970       1971       1972       1973       1974      1975  ...  2060  \
0        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
1        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
2        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
3        NaN        NaN        NaN        NaN        NaN       NaN  ...   NaN   
4  54.822121  54.894138  56.209438  57.267109  57.991138  59.36554  ...   NaN   

   2065  2070  2075  2080  2085  2090  2095  2100  Unnamed: 69  
0   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
3   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
4   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[5 rows x 70 columns]

> Data infos

In [85]:
dfEdStatsData.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 886930 entries, 0 to 886929
Data columns (total 70 columns):
 #   Column          Non-Null Count   Dtype  
---  ------          --------------   -----  
 0   Country Name    886930 non-null  object 
 1   Country Code    886930 non-null  object 
 2   Indicator Name  886930 non-null  object 
 3   Indicator Code  886930 non-null  object 
 4   1970            72288 non-null   float64
 5   1971            35537 non-null   float64
 6   1972            35619 non-null   float64
 7   1973            35545 non-null   float64
 8   1974            35730 non-null   float64
 9   1975            87306 non-null   float64
 10  1976            37483 non-null   float64
 11  1977            37574 non-null   float64
 12  1978            37576 non-null   float64
 13  1979            36809 non-null   float64
 14  1980            89122 non-null   float64
 15  1981            38777 non-null   float64
 16  1982            37511 non-null   float64
 17  1983      

Our dataset contains 886930 entries, we have 66 float variables and 4 object variables which are strings but we keep them in object type as they are handled better like this.


## Dataset fractioning

We create sub datasets for "country-only" and "worldwide"

In [86]:
# we convert the column names to serpent case
dfEdStatsData.columns = dfEdStatsData.columns.str.lower().str.replace(' ', '_')
dfEdStatsData

# world sub dataset
dfEdStatsDataWorld = dfEdStatsData[dfEdStatsData['country_name'] == 'World']

# country sub dataset
# we check for the first occurence of Afghanistan as it is the first country in the dataset
firstOccurence = dfEdStatsData[dfEdStatsData['country_name'] == 'Afghanistan'].index[0]
# we split the dataset into two parts
dfEdStatsDataCountry = dfEdStatsData.iloc[firstOccurence:]
dfEdStatsDataCountry.reset_index(drop=True, inplace=True)
dfEdStatsDataCountry

country_name country_code  \
0       Afghanistan          AFG   
1       Afghanistan          AFG   
2       Afghanistan          AFG   
3       Afghanistan          AFG   
4       Afghanistan          AFG   
...             ...          ...   
795300     Zimbabwe          ZWE   
795301     Zimbabwe          ZWE   
795302     Zimbabwe          ZWE   
795303     Zimbabwe          ZWE   
795304     Zimbabwe          ZWE   

                                           indicator_name  \
0       Adjusted net enrolment rate, lower secondary, ...   
1       Adjusted net enrolment rate, lower secondary, ...   
2       Adjusted net enrolment rate, lower secondary, ...   
3       Adjusted net enrolment rate, lower secondary, ...   
4       Adjusted net enrolment rate, primary, both sex...   
...                                                   ...   
795300  Youth illiterate population, 15-24 years, male...   
795301  Youth literacy rate, population 15-24 years, b...   
795302  Youth literacy rate, population 15-24 years, f...   
795303  Youth literacy rate, population 15-24 years, g...   
795304  Youth literacy rate, population 15-24 years, m...   

              indicator_code  1970  1971  1972  1973      1974  1975  ...  \
0                 UIS.NERA.2   NaN   NaN   NaN   NaN   7.05911   NaN  ...   
1               UIS.NERA.2.F   NaN   NaN   NaN   NaN   2.53138   NaN  ...   
2             UIS.NERA.2.GPI   NaN   NaN   NaN   NaN   0.22154   NaN  ...   
3               UIS.NERA.2.M   NaN   NaN   NaN   NaN  11.42652   NaN  ...   
4                SE.PRM.TENR   NaN   NaN   NaN   NaN       NaN   NaN  ...   
...                      ...   ...   ...   ...   ...       ...   ...  ...   
795300      UIS.LP.AG15T24.M   NaN   NaN   NaN   NaN       NaN   NaN  ...   
795301     SE.ADT.1524.LT.ZS   NaN   NaN   NaN   NaN       NaN   NaN  ...   
795302  SE.ADT.1524.LT.FE.ZS   NaN   NaN   NaN   NaN       NaN   NaN  ...   
795303  SE.ADT.1524.LT.FM.ZS   NaN   NaN   NaN   NaN       NaN   NaN  ...   
795304  SE.ADT.1524.LT.MA.ZS   NaN   NaN   NaN   NaN       NaN   NaN  ...   

        2060  2065  2070  2075  2080  2085  2090  2095  2100  unnamed:_69  
0        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
1        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
2        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
3        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
4        NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
...      ...   ...   ...   ...   ...   ...   ...   ...   ...          ...  
795300   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
795301   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
795302   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
795303   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  
795304   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN   NaN          NaN  

[795305 rows x 70 columns]

## Null percentage by indicator

Ranking of most relevant indicators depending on null percentage for data aggregated **in the past**

In [87]:
dfNullPercentageByIndicator = dfEdStatsDataCountry.groupby('indicator_code').apply(lambda x: x.isnull().mean())
dfNullPercentageByIndicator

# get 1970 column id
year1970 = dfEdStatsDataCountry.columns.get_loc('1970')
# get 2016 column id
year2016 = dfEdStatsDataCountry.columns.get_loc('2016')

# we get column between 1970 and 2016 using iloc
dfNullPercentageByIndicatorPast = dfNullPercentageByIndicator.iloc[:, year1970:year2016+1]
dfNullPercentageByIndicatorPast

# we get the mean of the null values for each indicator
dfNullPercentageByIndicatorPast['mean'] = dfNullPercentageByIndicatorPast.mean(axis=1)
dfNullPercentageByIndicatorPast = dfNullPercentageByIndicatorPast.sort_values('mean', ascending=True)
dfNullPercentageByIndicatorPast

C:\Users\alexa\AppData\Local\Temp\ipykernel_4164\2263488496.py:1: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  dfNullPercentageByIndicator = dfEdStatsDataCountry.groupby('indicator_code').apply(lambda x: x.isnull().mean())
C:\Users\alexa\AppData\Local\Temp\ipykernel_4164\2263488496.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfNullPercentageByIndicatorPast['mean'] = dfNullPercentageByIndicatorPast.mean(axis=1)


1970      1971      1972      1973      1974  \
indicator_code                                                           
SP.POP.TOTL           0.027650  0.027650  0.027650  0.027650  0.027650   
SP.POP.GROW           0.027650  0.027650  0.027650  0.027650  0.027650   
SE.PRM.DURS           0.073733  0.073733  0.073733  0.073733  0.073733   
SE.PRM.AGES           0.073733  0.073733  0.073733  0.073733  0.073733   
UIS.THDUR.0           0.078341  0.073733  0.073733  0.073733  0.073733   
...                        ...       ...       ...       ...       ...   
SABER.TER.GOAL6.LVL1  1.000000  1.000000  1.000000  1.000000  1.000000   
SABER.TER.GOAL6.LVL2  1.000000  1.000000  1.000000  1.000000  1.000000   
SABER.TER.GOAL6.LVL3  1.000000  1.000000  1.000000  1.000000  1.000000   
SABER.TER.GOAL4       1.000000  1.000000  1.000000  1.000000  1.000000   
SABER.TER.GOAL3       1.000000  1.000000  1.000000  1.000000  1.000000   

                          1975      1976      1977      1978      1979  ...  \
indicator_code                                                          ...   
SP.POP.TOTL           0.027650  0.027650  0.027650  0.027650  0.027650  ...   
SP.POP.GROW           0.027650  0.027650  0.027650  0.027650  0.027650  ...   
SE.PRM.DURS           0.073733  0.073733  0.073733  0.073733  0.073733  ...   
SE.PRM.AGES           0.073733  0.073733  0.073733  0.073733  0.073733  ...   
UIS.THDUR.0           0.073733  0.073733  0.073733  0.073733  0.073733  ...   
...                        ...       ...       ...       ...       ...  ...   
SABER.TER.GOAL6.LVL1  1.000000  1.000000  1.000000  1.000000  1.000000  ...   
SABER.TER.GOAL6.LVL2  1.000000  1.000000  1.000000  1.000000  1.000000  ...   
SABER.TER.GOAL6.LVL3  1.000000  1.000000  1.000000  1.000000  1.000000  ...   
SABER.TER.GOAL4       1.000000  1.000000  1.000000  1.000000  1.000000  ...   
SABER.TER.GOAL3       1.000000  1.000000  1.000000  1.000000  1.000000  ...   

                          2008      2009      2010      2011      2012  \
indicator_code                                                           
SP.POP.TOTL           0.009217  0.009217  0.009217  0.009217  0.013825   
SP.POP.GROW           0.009217  0.009217  0.009217  0.013825  0.013825   
SE.PRM.DURS           0.041475  0.041475  0.041475  0.041475  0.041475   
SE.PRM.AGES           0.041475  0.041475  0.041475  0.041475  0.041475   
UIS.THDUR.0           0.055300  0.055300  0.055300  0.055300  0.055300   
...                        ...       ...       ...       ...       ...   
SABER.TER.GOAL6.LVL1  1.000000  1.000000  1.000000  1.000000  1.000000   
SABER.TER.GOAL6.LVL2  1.000000  1.000000  1.000000  1.000000  1.000000   
SABER.TER.GOAL6.LVL3  1.000000  1.000000  1.000000  1.000000  1.000000   
SABER.TER.GOAL4       1.000000  1.000000  1.000000  1.000000  1.000000   
SABER.TER.GOAL3       1.000000  1.000000  1.000000  1.000000  1.000000   

                          2013      2014      2015      2016      mean  
indicator_code                                                          
SP.POP.TOTL           0.013825  0.032258  0.032258  0.032258  0.020590  
SP.POP.GROW           0.013825  0.032258  0.032258  0.032258  0.021179  
SE.PRM.DURS           0.041475  0.041475  0.041475  0.179724  0.055888  
SE.PRM.AGES           0.041475  0.041475  0.041475  0.179724  0.055888  
UIS.THDUR.0           0.055300  0.055300  0.055300  0.193548  0.060888  
...                        ...       ...       ...       ...       ...  
SABER.TER.GOAL6.LVL1  1.000000  1.000000  1.000000  1.000000  1.000000  
SABER.TER.GOAL6.LVL2  1.000000  1.000000  1.000000  1.000000  1.000000  
SABER.TER.GOAL6.LVL3  1.000000  1.000000  1.000000  1.000000  1.000000  
SABER.TER.GOAL4       1.000000  1.000000  1.000000  1.000000  1.000000  
SABER.TER.GOAL3       1.000000  1.000000  1.000000  1.000000  1.000000  

[3665 rows x 48 columns]

Ranking of most relevant indicators depending on null percentage for data **expected in the future**

In [88]:
# get 2020 column id
year2020 = dfNullPercentageByIndicator.columns.get_loc('2020')
# get 2100 column id
year2100 = dfNullPercentageByIndicator.columns.get_loc('2100')

# we get column between 2017 and 2100 using iloc
dfNullPercentageByIndicatorFuture = dfNullPercentageByIndicator.iloc[:, year2020:year2100+1]
dfNullPercentageByIndicatorFuture

dfNullPercentageByIndicatorFuture['mean'] = dfNullPercentageByIndicatorFuture.mean(axis=1)
dfNullPercentageByIndicatorFuture = dfNullPercentageByIndicatorFuture.sort_values('mean', ascending=True)
dfNullPercentageByIndicatorFuture

C:\Users\alexa\AppData\Local\Temp\ipykernel_4164\843556840.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dfNullPercentageByIndicatorFuture['mean'] = dfNullPercentageByIndicatorFuture.mean(axis=1)


2020      2025      2030      2035      2040  \
indicator_code                                                                  
PRJ.ATT.60UP.1.MF            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.FE            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.MA            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.MF            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.NED.FE          0.235023  0.235023  0.235023  0.235023  0.235023   
...                               ...       ...       ...       ...       ...   
LO.LLECE.SCI6.3              1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.SCI6.3.FE           1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.SCI6.3.MA           1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.REA6.3.FE           1.000000  1.000000  1.000000  1.000000  1.000000   
XGDP.56.FSGOV.FDINSTADM.FFD  1.000000  1.000000  1.000000  1.000000  1.000000   

                                 2045      2050      2055      2060      2065  \
indicator_code                                                                  
PRJ.ATT.60UP.1.MF            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.FE            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.MA            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.MF            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.NED.FE          0.235023  0.235023  0.235023  0.235023  0.235023   
...                               ...       ...       ...       ...       ...   
LO.LLECE.SCI6.3              1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.SCI6.3.FE           1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.SCI6.3.MA           1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.REA6.3.FE           1.000000  1.000000  1.000000  1.000000  1.000000   
XGDP.56.FSGOV.FDINSTADM.FFD  1.000000  1.000000  1.000000  1.000000  1.000000   

                                 2070      2075      2080      2085      2090  \
indicator_code                                                                  
PRJ.ATT.60UP.1.MF            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.FE            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.MA            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.4.MF            0.235023  0.235023  0.235023  0.235023  0.235023   
PRJ.ATT.2064.NED.FE          0.235023  0.235023  0.235023  0.235023  0.235023   
...                               ...       ...       ...       ...       ...   
LO.LLECE.SCI6.3              1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.SCI6.3.FE           1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.SCI6.3.MA           1.000000  1.000000  1.000000  1.000000  1.000000   
LO.LLECE.REA6.3.FE           1.000000  1.000000  1.000000  1.000000  1.000000   
XGDP.56.FSGOV.FDINSTADM.FFD  1.000000  1.000000  1.000000  1.000000  1.000000   

                                 2095      2100      mean  
indicator_code                                             
PRJ.ATT.60UP.1.MF            0.235023  0.235023  0.235023  
PRJ.ATT.2064.4.FE            0.235023  0.235023  0.235023  
PRJ.ATT.2064.4.MA            0.235023  0.235023  0.235023  
PRJ.ATT.2064.4.MF            0.235023  0.235023  0.235023  
PRJ.ATT.2064.NED.FE          0.235023  0.235023  0.235023  
...                               ...       ...       ...  
LO.LLECE.SCI6.3              1.000000  1.000000  1.000000  
LO.LLECE.SCI6.3.FE           1.000000  1.000000  1.000000  
LO.LLECE.SCI6.3.MA           1.000000  1.000000  1.000000  
LO.LLECE.REA6.3.FE           1.000000  1.000000  1.000000  
XGDP.56.FSGOV.FDINSTADM.FFD  1.000000  1.000000  1.000000  

[3665 rows x 18 columns]

## Main indicators ranking

Population | SP.POP.1524.TO.UN

Economique | NY.GNP.PCAP.PP.CD

Education | SE.SEC.ENRR | SE.TER.ENRR

Numerique | IT.NET.USER.P2

In [89]:
numberOfIndicators= dfNullPercentageByIndicatorPast.shape[0]

# population indicator rank

populationIndicatorRankPast = dfNullPercentageByIndicatorPast.index.get_loc('SP.POP.TOTL')

populationIndicatorRankFuture = dfNullPercentageByIndicatorFuture.index.get_loc('SP.POP.TOTL')

# economic indicator rank

economicIndicatorRankPast = dfNullPercentageByIndicatorPast.index.get_loc('NY.GNP.PCAP.PP.CD')

economicIndicatorRankFuture = dfNullPercentageByIndicatorFuture.index.get_loc('NY.GNP.PCAP.PP.CD')

# eduction first indicator rank

educationIndicatorRankPast = dfNullPercentageByIndicatorPast.index.get_loc('SE.SEC.ENRR')

educationIndicatorRankFuture = dfNullPercentageByIndicatorFuture.index.get_loc('SE.SEC.ENRR')

# eduction second indicator rank

educationIndicatorRankPast2 = dfNullPercentageByIndicatorPast.index.get_loc('SE.TER.ENRR')

educationIndicatorRankFuture2 = dfNullPercentageByIndicatorFuture.index.get_loc('SE.TER.ENRR')

# numeric indicator rank

numericIndicatorRankPast = dfNullPercentageByIndicatorPast.index.get_loc('IT.NET.USER.P2')

numericIndicatorRankFuture = dfNullPercentageByIndicatorFuture.index.get_loc('IT.NET.USER.P2')

print("Past aggregated data rankings :\n")

print('Rank of SP.POP.TOTL in the list of indicators: '+ str(populationIndicatorRankPast) +'/'+ str(numberOfIndicators))

print('Rank of NY.GNP.PCAP.PP.CD in the list of indicators: '+ str(economicIndicatorRankPast) +'/'+ str(numberOfIndicators))

print('Rank of SE.SEC.ENRR in the list of indicators: '+ str(educationIndicatorRankPast) +'/'+ str(numberOfIndicators))

print('Rank of SE.TER.ENRR in the list of indicators: '+ str(educationIndicatorRankPast2) +'/'+ str(numberOfIndicators))

print('Rank of IT.NET.USER.P2 in the list of indicators: '+ str(numericIndicatorRankPast) +'/'+ str(numberOfIndicators))

print("\nFuture aggregated data rankings :\n")

print('Rank of SP.POP.TOTL in the list of indicators: '+ str(populationIndicatorRankFuture) +'/'+ str(numberOfIndicators))

print('Rank of NY.GNP.PCAP.PP.CD in the list of indicators: '+ str(economicIndicatorRankFuture) +'/'+ str(numberOfIndicators))

print('Rank of SE.SEC.ENRR in the list of indicators: '+ str(educationIndicatorRankFuture) +'/'+ str(numberOfIndicators))

print('Rank of SE.TER.ENRR in the list of indicators: '+ str(educationIndicatorRankFuture2) +'/'+ str(numberOfIndicators))

print('Rank of IT.NET.USER.P2 in the list of indicators: '+ str(numericIndicatorRankFuture) +'/'+ str(numberOfIndicators))

Past aggregated data rankings :

Rank of SP.POP.TOTL in the list of indicators: 0/3665
Rank of NY.GNP.PCAP.PP.CD in the list of indicators: 134/3665
Rank of SE.SEC.ENRR in the list of indicators: 72/3665
Rank of SE.TER.ENRR in the list of indicators: 97/3665
Rank of IT.NET.USER.P2 in the list of indicators: 132/3665

Future aggregated data rankings :

Rank of SP.POP.TOTL in the list of indicators: 1757/3665
Rank of NY.GNP.PCAP.PP.CD in the list of indicators: 3092/3665
Rank of SE.SEC.ENRR in the list of indicators: 379/3665
Rank of SE.TER.ENRR in the list of indicators: 667/3665
Rank of IT.NET.USER.P2 in the list of indicators: 2034/3665
